.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: CI
+
+on:
+  push:
+    branches: [ main, staging, trying ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  CARGO_TERM_COLOR: always
+  TARGET: thumbv7em-none-eabihf
+  HAL: nrf52840-hal = "0.12.0"
+
+# NOTE if you change the `matrix` field or add/remove a job you'll need to update the list of jobs
+# in `.github/bors.toml
+jobs:
+  test:
+    strategy:
+      matrix:
+        rust:
+          - stable
+          - nightly
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: app-template
+    - uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ matrix.rust }}
+        override: true
+        target: ${{ env.TARGET }}
+    - name: debug
+      run: cd app-template && git status && ls -la
+    - name: Install cargo-generate
+      run: cargo install cargo-generate --debug
+    - name: Instantiate template
+      run: |
+        git config --global init.defaultbranch main
```

Comment:
could you also try to git configure:
```
git config --global user.name "Your Name"
git config --global user.email "youremail@yourdomain.com"
```

IIRC that causes once for me on a ci run also a problem. See our [workflow config](https://github.com/cargo-generate/cargo-generate/blob/master/.github/workflows/build.yml#L80)

Reply:
Hmm, that didn't seem to do anything here